src/middlewared/middlewared/plugins/vm.py

```diff
@@ -87,7 +88,26 @@ class VMSupervisor(object):
 
     def __init__(self, manager, vm):
         self.manager = manager
-        self.logger = self.manager.logger
+
+        if not os.path.exists('/var/log/vm'):
+            os.makedirs('/var/log/vm')
+
+        self.logger = middlewared.logger.Logger(
```

Comment:
To ensure loggers hierarchy we usually do it like this `self.manager.logger.getChild(f'vm_{vm["id"]}')`. Also we usually don't put human-readable things like VM names (that can contain whitespaces, punctuation and other stuff) to logger names because they are identificators.